### ISSUE TITLE
`io.micronaut.core.annotation.Nullable` has stereotype `@Nonnull(when = When.MAYBE)` on it hence fails on null properties supplied

### ISSUE THREAD

**Issue by soberich** [micronaut-projects/micronaut-data#1365]
### Expected Behavior

`io.micronaut.core.annotation.Nullable` should make parameter non-required, nullable.

### Actual Behaviour

This on is kind of nonsense... so annotation `io.micronaut.core.annotation.Nullable` has `@Nonnull(when = When.MAYBE)`, so this module - Serde - makes annotated properties required, non-nullable 🥇 
:)
When I removed annotation `io.micronaut.core.annotation.Nullable` after several hours of trials and failures- it worked!

`io.micronaut.core.annotation.AnnotatedElement`
```java
    /**
     * @return Whether the element is declared as not being null
     */
    default boolean isNonNull() { // this gives `true` for property annotated with `io.micronaut.core.annotation.Nullable` 
        return getAnnotationMetadata().hasStereotype(AnnotationUtil.NON_NULL);
    }
```

### Steps To Reproduce

I guess it simple enough from description.

### Environment Information

Serde m7
micronaut 3.3.3

### Example Application

https://github.com/soberich/micronaut-serialization-134

### Version

3.3.3

**Comment by yawkat** [comment]
I can reproduce it with your test case @soberich, thank you. It is very similar to the steps described in https://github.com/micronaut-projects/micronaut-core/issues/6795 .

This time it is caused by this mapper: https://github.com/micronaut-projects/micronaut-data/blob/master/data-processor/src/main/java/io/micronaut/data/processor/mappers/jpa/jx/GeneratedValueMapper.java

The GeneratedValueMapper returns the additional annotation `javax.annotation.Nullable`, which is in turn annotated with the `@Nonnull(when = When.MAYBE)` annotation, which leads to the introduction of the stereotype. An additional complication is that this only happens if `javax.annotation.Nullable` is on the classpath, which is introduced in the reproducer by the guava dependency.

@graemerocher how should we fix this? Unlike the `NullableAnnotationMapper`, we can't just remove the `GeneratedValueMapper`. I think the best way is to actually fix https://github.com/micronaut-projects/micronaut-core/issues/6795 in core.

Test case for this issue (but remember it requires guava on the classpath to fail):

```groovy
    void "test @Nullable with @GeneratedValue"() {
        when:
        def pojo = buildBeanIntrospection('test.Pojo', '''
package test;

import io.micronaut.core.annotation.Introspected;
import io.micronaut.core.annotation.Nullable;

@Introspected
public final class Pojo {
    @Nullable
    @jakarta.persistence.GeneratedValue // <- N.B.
    private final String surname;

    public Pojo(@Nullable String surname) {
        this.surname = surname;
    }

    @Nullable
    public String getSurname() {
        return this.surname;
    }
}
''')
        then:
        !pojo.getProperty("surname").get().isNonNull()
    }
```

**Comment by graemerocher** [comment]
yes I think we should fix the core issue

**Comment by soberich** [comment]
@yawkat pardon I thought it's visible when I reference this issue in commit message.. Probably, it's one of my chrome extensions.
<img width="923" alt="image" src="https://user-images.githubusercontent.com/25544967/156348423-bc2ef577-fd49-40bd-9347-a1fdddb9e0d9.png">

**Comment by yawkat** [comment]
Fixed in core 3.3.x by https://github.com/micronaut-projects/micronaut-core/pull/6980 . thanks for the report.
